### ISSUE TITLE
Non-chief replicas freeze after chief completes training

### ISSUE THREAD

**Issue by c8h10n4o2ed** [tensorflow/tensorflow#16213]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 16.04
- **TensorFlow installed from (source or binary)**: Binary
- **TensorFlow version (use command below)**: v1.4.0-19-ga52c8d9, 1.4.1
- **Python version**: 2.7
- **Bazel version (if compiling from source)**:
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**: 6.0
- **GPU model and memory**: Titan X (Pascal), 12 GiB
- **Exact command to reproduce**: Custom Script

### Describe the problem
Non-chief replicas freeze after chief completes training, when in synchronous mode.
The problem appears to occur immediately after the chief shuts down.

See attached source code for a complete example of multi-GPU demonstrating the problem on a toy dataset. Modify the cluster variable and start the PS first, followed by workers then the chief node last. This is somewhat broken out in run_distributed.sh.

``
        sv = tf.train.Supervisor(
            is_chief=(FLAGS.task_index == 0),
            global_step = global_step,
            init_op = init_op
        )
        ...
        with sv.prepare_or_wait_for_session(server.target, config=config) as sess:
            # is chief
            if FLAGS.task_index == 0:
                sv.start_queue_runners(sess, [chief_queue_runner])
                sess.run(init_token_op)
        ...
``

### Source code / logs
[rnn-multi-gpu.zip](https://github.com/tensorflow/tensorflow/files/1641799/rnn-multi-gpu.zip)

**Comment by tatatodd** [comment]
Maybe @mrry or @suharshs have suggestions, or know others who might have suggestions.

**Comment by tensorflowbutler** [comment]
Nagging Awaiting TensorFlower: It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by mrry** [comment]
I think the answer here is to use an external solution (e.g. a cluster manager like Kubernetes or Mesos) to notice that the chief has terminated and bring down the other replicas. As far as I know there is no signaling built into the `tf.train.Supervisor` or its descendants (`tf.train.MonitoredSession`, `tf.estimator.Estimator`) that will cause the non-chief replicas to exit reliably.

/cc @ispirmustafa in case he knows of any approaches that might work with `tf.train.MonitoredSession` or `tf.estimator.Estimator`.

**Comment by tensorflowbutler** [comment]
Nagging Awaiting Response: It has been 14 days with no activityand the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by jart** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
